pxf/pxf-json/src/test/java/org/apache/hawq/pxf/plugins/json/JsonExtensionTest.java

```diff
@@ -0,0 +1,288 @@
+		columnDefs.add(new Pair<String, DataType>("entities.hashtags[0]", DataType.TEXT));
+		columnDefs.add(new Pair<String, DataType>("coordinates.coordinates[0]", DataType.FLOAT8));
+		columnDefs.add(new Pair<String, DataType>("coordinates.coordinates[1]", DataType.FLOAT8));
+	}
+
+	@After
+	public void cleanup() throws Exception {
+		columnDefs.clear();
+		extraParams.clear();
+	}
+
+	@Test
+	public void testCompressedMultilineJsonFile() throws Exception {
+
+		extraParams.add(new Pair<String, String>("IDENTIFIER", "created_at"));
+
+		List<String> output = new ArrayList<String>();
+
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547115253761,text1,SpreadButter,tweetCongress,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547123646465,text2,patronusdeadly,,,");
+		output.add("Fri Jun 07 22:45:02 +0000 2013,343136547136233472,text3,NoSecrets_Vagas,,,");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/tweets.tar.gz"), output);
+	}
+
+	@Test
+	public void testMaxRecordLength() throws Exception {
+
+		// variable-size-objects.json contains 3 json objects but only 2 of them fit in the 27 byte length limitation
+
+		extraParams.add(new Pair<String, String>("IDENTIFIER", "key666"));
+		extraParams.add(new Pair<String, String>("MAXLENGTH", "27"));
+
+		columnDefs.clear();
+		columnDefs.add(new Pair<String, DataType>("key666", DataType.TEXT));
+
+		List<String> output = new ArrayList<String>();
+
+		output.add("small object1");
+		// skip the large object2 XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+		output.add("small object3");
+
+		super.assertOutput(new Path(System.getProperty("user.dir") + File.separator
+				+ "src/test/resources/variable-size-objects.json"), output);
+	}
+
+	@Test
+	public void testDataTypes() throws Exception {
+
+		// TDOO: The BYTEA type is not tested!!! Current oneField.val = val.asText().getBytes(); convention is
+		// disputable!
+
+		extraParams.clear();
+		extraParams.add(new Pair<String, String>("IDENTIFIER", "bintType"));
+
+		columnDefs.clear();
+		columnDefs.add(new Pair<String, DataType>("bintType", DataType.BIGINT));
+		columnDefs.add(new Pair<String, DataType>("booleanType", DataType.BOOLEAN));
+		columnDefs.add(new Pair<String, DataType>("charType", DataType.CHAR));
+		// columnDefs.add(new Pair<String, DataType>("byteaType", DataType.BYTEA));
+		columnDefs.add(new Pair<String, DataType>("float8Type", DataType.FLOAT8));
+		columnDefs.add(new Pair<String, DataType>("realType", DataType.REAL));
+		columnDefs.add(new Pair<String, DataType>("integerType", DataType.INTEGER));
+		columnDefs.add(new Pair<String, DataType>("smallintType", DataType.SMALLINT));
+		columnDefs.add(new Pair<String, DataType>("bpcharType", DataType.BPCHAR));
+		columnDefs.add(new Pair<String, DataType>("varcharType", DataType.VARCHAR));
+		columnDefs.add(new Pair<String, DataType>("text", DataType.TEXT));
```

Comment:
perhaps add a comment saying this field doesn't exist in the file?
btw, does the order have to match between the table definition (columnDefs) and the json file itself? In other words, if the order between bintType and booleanType will be replaced here, will it still parse the file correctly?